pilot/pkg/proxy/envoy/v2/ads.go

```diff
@@ -522,18 +527,18 @@ func (s *DiscoveryServer) StreamAggregatedResources(stream ads.AggregatedDiscove
 				}
 
 			case EndpointType:
-				clusters := discReq.GetResourceNames()
 				if discReq.ErrorDetail != nil {
 					adsLog.Warnf("ADS:EDS: ACK ERROR %v %s %v", peerAddr, con.ConID, discReq.String())
 					edsReject.With(prometheus.Labels{"node": discReq.Node.Id, "err": discReq.ErrorDetail.Message}).Add(1)
 					totalXDSRejects.Add(1)
+					continue
 				}
-
+				clusters := discReq.GetResourceNames()
 				if clusters == nil && discReq.ResponseNonce != "" {
 					// There is no requirement that ACK includes clusters. The test doesn't.
-					if discReq.ErrorDetail == nil && discReq.ResponseNonce != "" {
-						con.EndpointNonceAcked = discReq.ResponseNonce
-					}
+					con.mu.Lock()
```

Comment:
Same - AFAIK NonceAcked is not the same thing with nonce nacked. It's a semantic change

Reply:
I do not change this semantic , L530-L534 already guaranteed NACK is returned. 
If run into here, it must be ACK